etna/models/base.py

```diff
@@ -450,4 +467,303 @@ def get_model(self) -> Any:
+        """Fit Model.
+
+        Parameters
+        ----------
+        ts:
+            TSDataset with features
+
+        Returns
+        -------
+        :
+            Model after fit
+        """
+        ts_torch = ts.to_torch_dataset(self.make_samples)
+        self.raw_fit(ts_torch)
+        return self
+
+    def raw_fit(self, dataset: "Dataset") -> "DeepBaseModel":
+        """Fit model on torch like Dataset.
+
+        Parameters
+        ----------
+        dataset: Dataset
+            Torch like dataset for model fit
+
+        Returns
+        -------
+        :
+            Model after fit
+        """
+        if self.split_kwargs:
+            if isinstance(dataset, Sized):
+                tsdataset_length = len(dataset)
+            else:
+                tsdataset_length = self.split_kwargs["dataset_size"]
+            train_frac = self.split_kwargs["train_frac"]
+            train_dataset, val_dataset = random_split(
+                dataset,
+                lengths=[int(train_frac * tsdataset_length), tsdataset_length - int(train_frac * tsdataset_length)],
+                generator=self.split_kwargs.get("generator"),
+            )
+            train_dataloader = DataLoader(
+                train_dataset, batch_size=self.train_batch_size, shuffle=True, **self.train_dataloader_kwargs
+            )
+            val_dataloader: Optional[DataLoader] = DataLoader(
+                val_dataset, batch_size=self.test_batch_size, shuffle=False, **self.val_dataloader_kwargs
+            )
+        else:
+            train_dataloader = DataLoader(
+                dataset, batch_size=self.train_batch_size, shuffle=True, **self.train_dataloader_kwargs
+            )
+            val_dataloader = None
+
+        if "logger" not in self.trainer_kwargs:
+            self.trainer_kwargs["logger"] = tslogger.pl_loggers
+        else:
+            self.trainer_kwargs["logger"] += tslogger.pl_loggers
+
+        trainer = Trainer(**self.trainer_kwargs)
+        trainer.fit(self, train_dataloaders=train_dataloader, val_dataloaders=val_dataloader)
+        return self
+
+    def raw_predict(self, dataset: "Dataset") -> Dict[Tuple[str, str], np.ndarray]:
+        """Make inference on torch like Dataset.
+
+        Parameters
+        ----------
+        dataset: Dataset
+            Torch like dataset for model inference
+
+        Returns
+        -------
+        :
+            Dictionary with predictions
+        """
+        test_dataloader = DataLoader(
+            dataset, batch_size=self.test_batch_size, shuffle=False, **self.test_dataloader_kwargs
+        )
+
+        predictions_dict = dict()
+        for batch in test_dataloader:
+            segments = batch["segment"]
+            predictions = self(batch)
+            predictions_array = predictions.detach().numpy()
+            for idx, segment in enumerate(segments):
+                predictions_dict[(segment, "target")] = predictions_array[idx, :]
```

Comment:
Do we actually need tuple as key here if we hardcode "target"?

Reply:
I guess. 
For models which return samples or quantiles, we will have additional logic for quantiles generation. 
It could be another class, but it seems we will return more than `target` key from `forecast`

Reply:
May you create a draft for the issue to add base class for models with prediction intervals